### PR TITLE
Add CMake option to link MSVC runtime library statically

### PR DESCRIPTION
**Pull request by tbeu**

### CODE REVIEW

File: cmake/options.cmake [hunk 1]
```diff
@@ -17,6 +17,22 @@ option(MATIO_SHARED "Build shared matio library, disable for static library" ON)
 # Option to enable position-independent code (PIC)
 option(MATIO_PIC "Enable position-independent code (PIC), i.e., compilation with the -fPIC flag" ON)
 
+# Option to link the MSVC runtime library statically
+option(MATIO_STATIC_LINK_CRT "Link the MSVC runtime library statically" OFF)
+
+if(POLICY CMP0091)
+    # CMake >= 3.15 has CMAKE_MSVC_RUNTIME_LIBRARY to set the MSVCC runtime library
+    if(WIN32 AND CMAKE_GENERATOR MATCHES "Visual Studio .*|NMake .*")
```
Comment (MaartenBent) [inline cmake/options.cmake:25]: I think you can just use `if(MSVC)` here.

File: cmake/options.cmake [hunk 2]
```diff
@@ -17,6 +17,22 @@ option(MATIO_SHARED "Build shared matio library, disable for static library" ON)
...
+        if(MATIO_STATIC_LINK_CRT)
+            message(STATUS "Configuring to link the MSVC runtime library statically")
```
Comment (MaartenBent) [inline cmake/options.cmake:27]: Is it necessary to keep these messages?

File: cmake/options.cmake [hunk 3]
```diff
@@ -17,6 +17,22 @@ option(MATIO_SHARED "Build shared matio library, disable for static library" ON)
 # Option to enable position-independent code (PIC)
 option(MATIO_PIC "Enable position-independent code (PIC), i.e., compilation with the -fPIC flag" ON)
 
+# Option to link the MSVC runtime library statically
+option(MATIO_STATIC_LINK_CRT "Link the MSVC runtime library statically" OFF)
+
+if(POLICY CMP0091)
```
Comment (MaartenBent) [inline cmake/options.cmake:23]: Maybe move the option inside the policy check, so it is not visible in old CMake GUIs.
And also inside the `MSVC` check. Or do you plan to add something similar for other compilers too?

**Review by MaartenBent — COMMENTED** [review]
I only did a code review. I have not yet tried to build it.

File: cmake/options.cmake (hunk 2 above)
Comment (tbeu) [reply cmake/options.cmake]: Change to VERBOSE

**Comment by tbeu** [comment]
Because of dependencies it is hard to get it consistently right. Closing.

**Pull request closed without merging**